keycloak/keycloak-admin-ui
# User fed: Add Kerberos provider and tests

## Motivation
https://issues.redhat.com/browse/APPDUX-807

## Brief Description
Decision in the 2/2/21 keycloak status meeting was to move forward with functionality to add new providers directly using the 'Add new provider' dropdown in the User Federation settings view similar to the way it was done in the old admin console UI, instead of via the wizards, at least for now. 

Note that this PR also covers creating a new kerberos provider via the user fed empty state page as well, by clicking the 'Add Kerberos providers' card.

Also included in this PR is a suite of cypress tests for the kerberos user federation as a whole, which includes tests for:

- Creating kerberos provider from empty state
- Creating kerberos provider from the card view
- Changing an existing kerberos provider and saving (update)
- Changing an existing kerberos provider and cancelling (cancel)
- Disabling a kerberos provider
- Enabling a previously-disabled kerberos provider
- Deleting a kerberos provider using the card menu
- Deleting a kerberos provider using the settings view Action menu

## Verification Steps
1. Go to user federation with no kerberos providers configured.
2. Click the 'Add Kerberos providers' card and verify that you can successfully create a new kerberos provider using the form that follows.
3. With at least one kerberos provider configured, verify that you can successfully create a new kerberos provider using the Add new provider > Kerberos command.
4. Bonus points: run yarn start:cypress and verify that the kerberos provider tests all pass.

## Checklist:
